src/app/backend/handler/filter.go

```diff
@@ -177,3 +177,33 @@ func mapUrlToResource(url string) *string {
 	}
 	return &parts[3]
 }
+
+// remoteAddr extracts the remote address of the request, taking into
+// account proxy headers.
+func remoteAddr(r *http.Request) string {
+	if prior := r.Header.Get("X-Original-Forwarded-For"); prior != "" {
+		proxies := strings.Split(prior, ",")
+		if len(proxies) > 0 {
+			remoteAddr := strings.TrimSpace(proxies[0])
+			if remoteAddr != "" {
+				return remoteAddr
+			}
+		}
+	}
+
+	if prior := r.Header.Get("X-Forwarded-For"); prior != "" {
+		proxies := strings.Split(prior, ",")
```

Comment:
The part inside `if` statement is duplicated, please export it to the new function.

Reply:
fixed